pkg/inject/report.go

```diff
@@ -151,6 +153,14 @@ func (r *Report) disableByAnnotation(conf *ResourceConfig) (bool, string, string
 	// cli     | n/a       | ""       | yes      | false
 	// cli     | n/a       | disabled | no       | true
 
+	// Check https://github.com/linkerd/linkerd2/issues/3183
+	serviceAccountToken := conf.pod.spec.AutomountServiceAccountToken
+	if serviceAccountToken != nil {
+		if !*serviceAccountToken {
+			return true, disabledAutomountServiceAccountToken, ""
+		}
+	}
+
```

Comment:
I think this doesn't belong under the function `disableByAnnotation()`. Can you instead surface that as a new `Report` struct field, like `HostNetwork` and `Sidecar`? You could then set it in `newReport`, and use it in `Injectable()`.